File: app/src/main/java/com/otoniel/alura/ui/view/FormStudentView.java
package com.otoniel.alura.ui.view;

import com.otoniel.alura.dao.StudentDAO;
import com.otoniel.alura.model.Student;

public class FormStudentView {

    private final StudentDAO dao;
    private Student student;


    public void setStudent(Student student) {
        this.student = student;
    }

    public FormStudentView() {
        this.dao = new StudentDAO();
    }

    public Student getStudent() {
        return student;
    }

    public void save() {
        if (student.getId() == 0) {
            dao.save(student);
        } else {
            dao.edit(student);
        }
    }
}

File: app/src/main/java/com/otoniel/alura/ui/view/FormStudentView.kt
package com.otoniel.alura.ui.view

import android.content.Context
import com.otoniel.alura.model.Student
import com.otoniel.alura.room.StudentRoom
import com.otoniel.alura.room.dao.RoomStudentDAO


class FormStudentView(context: Context) {
    private val dao: RoomStudentDAO = StudentRoom.getInstance(context).getRoomStudentDAO()

    fun update(student: Student) {
        dao.update(student)
    }

    fun save(student: Student) {
        dao.save(student)
    }
}